### PR TITLE
Improve X-Types interoperability

### PR DESCRIPTION
**Pull request by richiware**
## Description

This PR brings:

- Changes in some IDL files to improve X-Types interoperability.
- Not taking into account the TypeInformation from other vendors.

Depends on:
- #4978
- eprosima/fast-dds-gen#361
- eprosima/fast-dds-docs#854

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- *N/A* Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl [hunk 1]
```diff
@@ -244,13 +244,19 @@ module DDS { module XTypes {
     struct ExtendedTypeDefn {
         // Empty. Available for future extension
     };
- 
 
-    
-    // The TypeIdentifier uniquely identifies a type (a set of equivalent 
+
+    // eProsima extension to support TypeIdentifier no value cases.
+    @final
```
Comment (MiguelCompany) [inline include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl:250]:
```suggestion
    @final @nested
```

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp [hunk 2]
```diff
@@ -111,19 +107,19 @@ class TypeLookupRequestListener : public fastdds::rtps::ReaderListener, public f
      * @brief Gets TypeObject from TypeObjectRegistry, creates and sends reply.
      * @param request_id[in] The SampleIdentity of the request.
      * @param request[in] The request data.
+     * @param vendor_id[in] Vendor identifier which sent the request.
```
Comment (MiguelCompany) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp:110]:
```suggestion
     * @param vendor_id[in] Vendor identifier that sent the request.
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl (hunk 1 above)
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl]: Done in e3b740df8

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp (hunk 2 above)
Comment (richiware) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp]: Done in e3b740df8

File: include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl [hunk 3]
```diff
@@ -247,7 +247,7 @@ module DDS { module XTypes {
 
 
     // eProsima extension to support TypeIdentifier no value cases.
-    @final
+    @final @nested
```
Comment (MiguelCompany) [inline include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl:250]: Code needs to be regenerated after this change.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl (hunk 3 above)
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/detail/dds-xtypes_typeobject.idl:250]: `@nested` is not supported currently by `fastddsgen`. The generated code is the same after this change.

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
There have been some failures which I don't know if they are related.

@richiware Would you mind rebasing just in case?

**Pull request merged**